src/arc.rs

```diff
@@ -183,47 +211,74 @@ impl PathArc {
     /// [`canonicalize`]: struct.PathAbs.html#method.canonicalize
     /// [`current_dir`]: fn.current_dir.html
     pub fn absolute(&self) -> Result<PathAbs> {
-        let mut components = self.components();
-        let mut stack: Vec<OsString> = Vec::new();
-
-        macro_rules! pop_stack { [] => {{
-            if let None = stack.pop() {
-                return Err(Error::new(
-                    io::Error::new(io::ErrorKind::NotFound, ".. consumed root"),
-                    "resolving absolute",
-                    self.clone(),
-                ));
+        let mut res = PathBuf::new();
+
+        fn maybe_init_res(res: &mut PathBuf, resolvee: &PathArc) -> Result<()> {
+            if res.as_os_str() != "" {
```

Comment:
nit: use `.is_empty()` instead.

Reply:
If `PathBuf` had a `.is_empty()` I would have used it; once I'd decided to fall back to comparing with the empty string, it didn't occur to me to check whether `OsStr::is_empty()` existed.

Fixed.